include/pick_ik/ik_memetic.hpp

```diff
@@ -0,0 +1,70 @@
+#pragma once
+
+#include <pick_ik/goal.hpp>
+#include <pick_ik/robot.hpp>
+
+#include <chrono>
+#include <memory>
+#include <moveit/robot_model/joint_model_group.h>
+#include <moveit/robot_model/robot_model.h>
+#include <optional>
+#include <random>
+#include <vector>
+
+namespace pick_ik {
+
+struct Individual {
+    std::vector<double> genes;  // Joint angles
+    double fitness;
+    double extinction;
+    std::vector<double> gradient;
+};
+
+class MemeticIk {
+   private:
+    std::vector<Individual> population_;
+
+    std::vector<double> best_;
+    double best_cost_;
+    std::vector<double> best_curr_;
+    double best_cost_curr_;
+
+    // Config params
+    size_t elite_count_ = 4;
+    size_t population_count_ = 16;
+
+    // RNG (TODO move out of here)
+    std::random_device rd_;
+    std::mt19937 gen_;
+    std::uniform_real_distribution<double> mutate_dist_{-0.5, 0.5};
+    std::uniform_real_distribution<double> uniform_dist_{0.0, 1.0};
```

Comment:
I sure can and want to.  Shall I tackle that in this PR or another?

Reply:
Either, just let me know which you'd prefer.

Reply:
Added into this PR, though @ChrisThrasher and I were just talking about something that came up here. If we're building RSL from source, we have to ensure that it's being built as a shared library (or that the `pick_ik_plugin` is static, which I'm not sure is something we want).

Once we have the RSL release binary, this should be fine.

Reply:
The way I see, building source dependencies as shared libraries is just mimicking what Debian releases already do since I believe all Debian-packaged libraries are build as shared libraries. No need to hardcode `-fPIC` in RSL.